index.js

```diff
@@ -292,15 +296,40 @@ let rcl = {
   },
 
   /**
-   * @param {Context} context - The context to be shutdown.
+   * Shuts down the given context by shutting down and destroying all nodes contained within.
+   * @param {Context} [context=Context.defaultContext()] - The context to be shutdown.
    * @return {undefined}
+   * @throws {Error} If the context is already shut down.
    */
   shutdown(context = Context.defaultContext()) {
     if (this.isShutdown(context)) {
       throw new Error('The module rclnodejs has been shutdown.');
-      return;
+    } else {
+      this._shutdown(context);
     }
+  },
 
+  /**
+   * Shuts down the given context by shutting down and destroying all nodes contained within.
+   * Does nothing if the context is already shut down.
+   * @param {Context} [context=Context.defaultContext()] - The context to be shutdown.
+   * @return {undefined}
+   */
+  tryShutdown(context = Context.defaultContext()) {
+    if (!this.isShutdown(context)) {
+      this._shutdown(context);
```

Comment:
This could still throw if the underlying rcl call fails. Also related to other `@throws` docs, they may throw for other reasons not specified in the doc.

Reply:
That's true. I wanted to *start* documenting it. It can be extended at any time.

Reply:
I suggest we keep the naming `shutdown` to align with that of `rclpy`and simply change the current `shutdown` method's behaviour to not throw exceptions when the context has been shut down (but may still throw exceptions thrown from `rcl`).

To indicate the case that the context is already shut down, we could use `debug()` to notify, how do you think @koonpeng @felixdivo?

Reply:
Sounds good.